filament/backend/src/metal/MetalHandles.mm

```diff
@@ -525,6 +516,30 @@ static MTLPixelFormat decidePixelFormat(id<MTLDevice> device, TextureFormat form
             texture = nil;
             break;
     }
+
+    // If swizzling is set, set up a swizzled texture view that we'll use when sampling this texture.
+    const bool isDefaultSwizzle =
+            r == TextureSwizzle::CHANNEL_0 &&
+            g == TextureSwizzle::CHANNEL_1 &&
+            b == TextureSwizzle::CHANNEL_2 &&
+            a == TextureSwizzle::CHANNEL_3;
+    // If texture is nil, then it must be a SAMPLER_EXTERNAL texture. We'll ignore this case for now.
+    // TODO: implement swizzling for external textures.
+    if (!isDefaultSwizzle && texture) {
+        if (@available(macOS 10.15, iOS 13, *)) {
```

Comment:
Maybe call `self->isTextureSwizzleSupported` here?  This ObjC stuff looks like a Harry Potter incantation to me.  :P

Reply:
So the compiler issues warnings when we use an Apple API newer than our iOS deployment target (iOS 11). The `@available` guard silences this warnings. The compiler isn't smart enough to propagate the check if we moved the `@available` check into a helper function, unfortunately.

Here's more information on [@available](https://clang.llvm.org/docs/LanguageExtensions.html#objective-c-available).